lib/templateCompilerModules/utils.ts

```diff
@@ -8,16 +8,46 @@ export interface ASTNode {
   attrs: Attr[]
 }
 
+import {UrlWithStringQuery, parse as uriParse} from 'url';
+
 export function urlToRequire (url: string): string {
+  let returnValue = `"${url}"`
   // same logic as in transform-require.js
   const firstChar = url.charAt(0)
   if (firstChar === '.' || firstChar === '~' || firstChar === '@') {
     if (firstChar === '~') {
       const secondChar = url.charAt(1)
       url = url.slice(secondChar === '/' ? 2 : 1)
     }
-    return `require("${url}")`
-  } else {
-    return `"${url}"`
+
+    const uriParts = parseUriParts(url)
+
+    !uriParts.hash
```

Comment:
Use `if` block and `return` early. Avoid using mutable variables.

Reply:
ok on if instead of ternary and return early;
not totally sure what is meant with mutable variables; pls comment on next commit